File: src/main/java/blockswarm/network/packets/BlockSitePacket.java
package blockswarm.network.packets;

import java.io.Serializable;

/**
 *
 * @author cal
 */
public class BlockSitePacket implements Serializable
{
    public final int version;
    public final byte[] signature, publicKey;
    public final String domain, filehash;

    public BlockSitePacket(String domain, String filehash, int version, byte[] signature, byte[] publicKey)
    {
        this.version = version;
        this.signature = signature;
        this.publicKey = publicKey;
        this.domain = domain;
        this.filehash = filehash;
    }
    
    public byte[] getData()
    {
        return (domain + ":" + filehash + ":" + version).getBytes();
    }
}

File: src/main/java/blockswarm/network/packets/BlockSitePacket.kt
package blockswarm.network.packets

import java.io.Serializable

/**
 *
 * @author cal
 */
class BlockSitePacket(val domain: String, val filehash: String, val version: Int, val signature: ByteArray, val publicKey: ByteArray) : Serializable {

    val data: ByteArray
        get() = "$domain:$filehash:$version".toByteArray()
}
